SPIFlashLogger.class.nut

```diff
@@ -76,53 +114,185 @@ class SPIFlashLogger {
-        local obj_remaining = obj_len;
+        local objPos = 0;
+        local objRemaining = objLength;
         do {
 
             // How far are we from the end of the sector
-            if (_at_pos < SPIFLASHLOGGER_SECTOR_META_SIZE) _at_pos = SPIFLASHLOGGER_SECTOR_META_SIZE;
-            local sec_remaining = SPIFLASHLOGGER_SECTOR_SIZE - _at_pos;
-            if (obj_remaining < sec_remaining) sec_remaining = obj_remaining;
+            if (_atPos < SPIFLASHLOGGER_SECTOR_METADATA_SIZE) _atPos = SPIFLASHLOGGER_SECTOR_METADATA_SIZE;
+            local secRemaining = SPIFLASHLOGGER_SECTOR_SIZE - _atPos;
+            if (objRemaining < secRemaining) secRemaining = objRemaining;
 
             // We are too close to the end of the sector, skip to the next sector
-            if (obj_pos == 0 && sec_remaining < SPIFLASHLOGGER_OBJECT_MIN_SIZE) {
-                _at_sec = (_at_sec + 1) % _sectors;
-                _at_pos = SPIFLASHLOGGER_SECTOR_META_SIZE;
+            if (objPos == 0 && secRemaining < SPIFLASHLOGGER_OBJECT_MIN_SIZE) {
+                _atSec = (_atSec + 1) % _sectors;
+                _atPos = SPIFLASHLOGGER_SECTOR_METADATA_SIZE;
             }
 
             // Now write the data
-            _write(obj, _at_sec, _at_pos, obj_pos, sec_remaining);
-            _map[_at_sec] = SPIFLASHLOGGER_SECTOR_DIRTY;
+            _write(obj, _atSec, _atPos, objPos, secRemaining);
+            _map[_atSec] = SPIFLASHLOGGER_SECTOR_DIRTY;
 
             // Update the positions
-            obj_pos += sec_remaining;
-            obj_remaining -= sec_remaining;
-            _at_pos += sec_remaining;
-            if (_at_pos >= SPIFLASHLOGGER_SECTOR_SIZE) {
-                _at_sec = (_at_sec + 1) % _sectors;
-                _at_pos = SPIFLASHLOGGER_SECTOR_META_SIZE;
+            objPos += secRemaining;
+            objRemaining -= secRemaining;
+            _atPos += secRemaining;
+            if (_atPos >= SPIFLASHLOGGER_SECTOR_SIZE) {
+                _atSec = (_atSec + 1) % _sectors;
+                _atPos = SPIFLASHLOGGER_SECTOR_METADATA_SIZE;
             }
-        } while (obj_remaining > 0);
+        } while (objRemaining > 0);
 
         _disable();
     }
 
+    // Reads objects from the logger synchronously,
+    // returning a single log object for the specified *index*.
+    //
+    // Returns:
+    // Deserialized object or null
+    //    - the most recent object when `index == -1`,
+    //    - the oldest object when `index == 1`,
+    //    - *null* when the value of *index* is greater than the number of logs,
+    //    - throws an exception when `index == 0`.
+    //
+    function readSync(index = 0) {
+        // Unexpected index value
+        if (index == 0)
+            throw "Invalid argument";
+
+        // identify first sector to read
+        local seekTo = math.abs(index);
+        local count = 0;
+        local i = 0;
+
+        // _atSec - indicates the current write position, therefore
```

Comment:
`_atSec` is current position or current sector? Something's confusing here.

Reply:
agree it should be sector